src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,702 @@
+                    AddLocalsInNestedScope(node, localsInNestedScope);
+
+                    var name = NameGenerator.GenerateUniqueName("unused",
+                        n => !usedNames.Contains(n) &&
+                             !localsInNestedScope.Contains(n) &&
+                             semanticModel.LookupSymbols(node.SpanStart, name: n).IsEmpty);
+                    usedNames.Add(name);
+                    return name;
+                }
+                finally
+                {
+                    localsInNestedScope.Free();
+                }
+            }
+
+            void AddLocalsInNestedScope(SyntaxNode node, PooledHashSet<string> localsInNestedScope)
+            {
+                var blockAncestor = node.FirstAncestorOrSelf<SyntaxNode>(n => syntaxFacts.IsExecutableBlock(n));
+                if (blockAncestor != null)
+                {
+                    foreach (var variableDeclarator in blockAncestor.DescendantNodes().OfType<TVariableDeclaratorSyntax>())
+                    {
+                        var name = syntaxFacts.GetIdentifierOfVariableDeclarator(variableDeclarator).ValueText;
+                        localsInNestedScope.Add(name);
+                    }
+                }
+            }
+        }
+
+        private void FixAll(
+            string diagnosticId,
+            IOrderedEnumerable<Diagnostic> diagnostics,
+            SemanticModel semanticModel,
+            SyntaxNode root,
+            UnusedValuePreference preference,
+            bool removeAssignments,
+            Func<SyntaxNode, string> generateUniqueNameAtSpanStart,
+            SyntaxEditor editor,
+            ISyntaxFactsService syntaxFacts,
+            CancellationToken cancellationToken)
+        {
+            switch (diagnosticId)
+            {
+                case IDEDiagnosticIds.ExpressionValueIsUnusedDiagnosticId:
+                    FixAllExpressionValueIsUnusedDiagnostics(diagnostics, semanticModel, root,
+                        preference, generateUniqueNameAtSpanStart, editor, syntaxFacts, cancellationToken);
+                    break;
+
+                case IDEDiagnosticIds.ValueAssignedIsUnusedDiagnosticId:
+                    FixAllValueAssignedIsUnusedDiagnostics(diagnostics, semanticModel, root,
+                        preference, removeAssignments, generateUniqueNameAtSpanStart, editor, syntaxFacts, cancellationToken);
+                    break;
+
+                default:
+                    throw ExceptionUtilities.Unreachable;
+            }
+        }
+
+        private void FixAllExpressionValueIsUnusedDiagnostics(
+            IOrderedEnumerable<Diagnostic> diagnostics,
+            SemanticModel semanticModel,
+            SyntaxNode root,
+            UnusedValuePreference preference,
+            Func<SyntaxNode, string> generateUniqueNameAtSpanStart,
+            SyntaxEditor editor,
+            ISyntaxFactsService syntaxFacts,
+            CancellationToken cancellationToken)
```

Comment:
I am stopping here for this class.  THe complexity level and interdependencies between functions and data just completely blew through my stack.